vyper/context/types/function.py

```diff
@@ -157,15 +184,24 @@ def from_FunctionDef(
         ContractFunctionType
         """
         kwargs: Dict[str, Any] = {}
-        if is_constant is not None:
-            kwargs["is_constant"] = is_constant
+        if is_immutable is not None:
+            kwargs["state_mutability"] = (
+                StateMutability.VIEW if is_immutable else StateMutability.NONPAYABLE
+            )
         if is_public is not None:
-            kwargs["is_public"] = is_public
+            kwargs["function_visibility"] = (
+                FunctionVisibility.PUBLIC if is_public else FunctionVisibility.PRIVATE
+            )
 
         # decorators
         for decorator in node.decorator_list:
 
             if isinstance(decorator, vy_ast.Call):
+                if "nonreentrant" in kwargs:
+                    raise StructureException(
+                        f"nonreentrant decorator is already set with key: {kwargs['nonreentrant']}",
+                        node,
+                    )
```

Comment:
I think this overrides a later check

Reply:
Yes, before if you specified `nonreentrant` multiple times, this would silently update. So, not really a big deal.